### DISCUSSION TITLE
How to read settings inside background app-service?

### DISCUSSION THREAD

**Discussion by mrklos** [zepp-health/zeppos-forum#388] [Q&A] [answered]
Sorry i cannot find an example, and debugging app-service is tricky...

**Comment by MyoungXUE** [comment]
The app-service example can be found at this link.

https://docs.zepp.com/docs/guides/framework/device/app-service/#full-example

**Reply by mrklos** [reply]
Thank You! But both:
https://github.com/zepp-health/zeppos-samples/blob/main/application/3.0/3.0-feature/app-service/time_service.js
and
https://github.com/zepp-health/zeppos-samples/blob/main/application/3.0/3.0-feature/app-service/system_event_service.js
do not use https://docs.zepp.com/docs/reference/app-settings-api/settings-storage/ or similiar

**Reply by mrklos** [reply, marked as answer]
Figured it out:

In AppSideService (using ZML library) 

    onSettingsChange({ key, newValue, oldValue }) {
      if (key === 'settings' && newValue) {
        settings = newValue ? JSON.parse(newValue) : DEFAULT_SETTINGS;
      }
    },
    onRequest(req, res) {
      if (req.method === 'GET_SETTINGS') {
        res(null, {
          result: settings
        })
      }
    },

and app service can call  this.request({
    method: 'GET_SETTINGS'
  })
